src/OutputCollector.jl

```diff
@@ -43,20 +43,26 @@ function LineStream(pipe::Pipe, event::Condition)
 
     lines = Tuple{Float64,String}[]
     task = @async begin
-        # Read lines in until we can't anymore
-        while !eof(pipe)
+        # Read lines in until we can't anymore.
+        while true
             # Push this line onto our lines, then notify() the event
             line = readuntil_many(pipe, ['\n', '\r'])
+            if isempty(line) && eof(pipe)
+                break
+            end
             push!(lines, (time(), line))
             notify(event)
         end
     end
 
+    # Switch to the task immediately so that we can start scooping up that
+    # sweet sweet data:
+    try yield(task) end
```

Comment:
This seems weird; what kind of error are you running into when `yield`-ing?

Reply:
Sometimes the task has already completed, so I can't `yield()`.